share/completions/argparse.fish

```diff
@@ -0,0 +1,12 @@
+set -g CONDITION '! __fish_seen_argument -s r --long required-val -s o --long optional-val'
```

Comment:
Same issue as in #8433 - this leaves the variable in-place. Please make it local instead.